pkgs/stdenv/cross/default.nix

```diff
@@ -0,0 +1,24 @@
+{ system, allPackages, platform, crossSystem, config, ... } @ args:
+
+rec {
+  argClobber = {
+    crossSystem = null;
+    # Ignore custom stdenvs when cross compiling for compatability
+    config = builtins.removeAttrs config [ "replaceStdenv" ];
+  };
+  vanillaStdenv = (import ../. (args // argClobber // {
+    allPackages = args: allPackages (argClobber // args);
+  })).stdenv;
+
+  # Yeah this isn't so cleanly just build-time packages yet. Notice the the
+  # buildPackages <-> stdenvCross cycle. Yup, it's very weird.
```

Comment:
This is working, because the derivation used to build the `stdenvCross` are in fact using `forceNativeDrv` to use the `nativeDrv` attribute of the resulting derivation build with `vanillaStdenv` (second argument of `makeStdenvCross`)  So technically you could use the same `buildPackages` as used in `custom.nix`.

I think you can fix this comment for now, and make a follow-up patch for that after.